R/checks.R

```diff
@@ -981,4 +981,48 @@ check_invalid_form1.1 <- function(form) {
         (ptstatus == 1 & is_invalid_na_or_n(condate)) |
         (ptstatus == 2 & is_invalid_or_na(condate))
     )
+}
+
+#' Time from injury to abx administration must be within the time interval (5h) from injury to hospital admission
+#' 
+#' @param merged_form a dataframe containing all forms
+#' @return a dataframe containing problematic entries with relevant columns
+#' @import tidyverse
+#' @export
+
+check_iahrs_ihhrs <- function(forms) {
+  problems <- select(forms,
+                     region, site, studyid, ptinit, locabx, ptstatus, admfrom, ihunits, ihhrs, ihdays, iaunits, iahrs, iadays) %>%
+    filter((locabx == 1 & ptstatus == 1 & ihunits == 1 & iaunits == 1 & ((iahrs < ihhrs) | (iahrs > ihhrs + 12))) 
```

Comment:
You don't have to do locabx == 1 & ptstatus == 1 everytime.
Put them outside of brackets